src/appshell/view/preferences/exportpreferencesmodel.cpp

```diff
@@ -0,0 +1,307 @@
+{
+    return exporter()->maxChannels();
+}
+
+QString ExportPreferencesModel::exportSampleRate() const
+{
+    auto currentSampleRate = exportConfiguration()->exportSampleRate();
+    for (const auto& rate : m_sampleRateMapping) {
+        if (currentSampleRate == rate.first) {
+            return rate.second;
+        }
+    }
+
+    return {};
+}
+
+QVariantList ExportPreferencesModel::exportSampleRateList()
+{
+    std::vector<int> sampleRateList = exporter()->sampleRateList();
+    if (sampleRateList.empty()) {
+        sampleRateList = DEFAULT_SAMPLE_RATE_LIST;
+    }
+
+    QVariantList result;
+    m_sampleRateMapping.clear();
+    for (const auto& rate : sampleRateList) {
+        QString sampleRateName = QString::number(rate) + " Hz";
+        m_sampleRateMapping.push_back(std::make_pair(rate, sampleRateName));
+        result << QVariant::fromValue(sampleRateName);
+    }
+
+    return result;
+}
+
+void ExportPreferencesModel::setExportSampleRate(const QString& rateName)
+{
+    if (rateName == exportSampleRate()) {
+        return;
+    }
+
+    auto it = std::find_if(m_sampleRateMapping.begin(), m_sampleRateMapping.end(),
+                           [&rateName](const auto& rate) { return rateName == rate.second; });
+    if (it != m_sampleRateMapping.end()) {
+        exportConfiguration()->setExportSampleRate(it->first);
+        emit exportSampleRateChanged();
+        return;
+    }
+}
+
+QString ExportPreferencesModel::exportSampleFormat() const
+{
+    NOT_IMPLEMENTED;
+    return {};
+}
+
+QVariantList ExportPreferencesModel::exportSampleFormatList() const
+{
+    NOT_IMPLEMENTED;
+    return {};
+}
+
+void ExportPreferencesModel::setExportSampleFormat(const QString& format)
+{
+    Q_UNUSED(format);
+    NOT_IMPLEMENTED;
+}
+
+void ExportPreferencesModel::updateCurrentSampleRate()
+{
+    int currentSampleRate = exportConfiguration()->exportSampleRate();
+    std::vector<int> sampleRateList = exporter()->sampleRateList();
+    for (const auto& rate : sampleRateList) {
+        if (rate == currentSampleRate) {
+            return;
+        }
+    }
+
+    //! NOTE: if current sample rate is not found within format's available sample rates
+    //! get the first one available
+    QVariantList stringSampleRateList = exportSampleRateList();
+    if (stringSampleRateList.empty()) {
+        return;
+    }
+    setExportSampleRate(stringSampleRateList[0].toString());
+}
+
+void ExportPreferencesModel::updateExportChannels()
+{
+    int maxChannels = exporter()->maxChannels();
+
+    if (static_cast<int>(exportChannels()) > maxChannels) {
+        setExportChannels(ExportChannelsPref::ExportChannels(maxChannels));
+    }
+}
+
+bool ExportPreferencesModel::verifyExportPossible()
+{
+    QDir directory(directoryPath());
+    if (!directory.exists() || directoryPath().isEmpty()) {
+        interactive()->error(muse::trc("export", "Export Audio"), muse::trc("export", "Unable to create destination folder"));
```

Comment:
hmm, but you’re not trying to create the folder - looks like you forgot to add makePath. And yes, please use FileSystem - filesystem operations should be mockable for tests.